clients/src/main/java/org/apache/kafka/common/requests/PushTelemetryRequest.java

```diff
@@ -71,6 +73,31 @@ public PushTelemetryRequestData data() {
         return data;
     }
 
+    public PushTelemetryResponse createResponse(int throttleTimeMs, Errors errors) {
+        PushTelemetryResponseData responseData = new PushTelemetryResponseData();
+        responseData.setErrorCode(errors.code());
+        responseData.setThrottleTimeMs(throttleTimeMs);
```

Comment:
> Could we redirect getErrorResponse to here and rename it to errorResponse?

Done

> If error code is not ThrottlingQuotaExceededException, we should ignore throttleTimeMs.

It might be naive but sorry I didn't understand as why throttleTimeMs should not be passed in response for other exceptions. Isn't all requests goes through common throttling code where requests might be throttled for sometime based on the throughput?

Reply:
Thanks for the explanation @junrao, this is helpful. I have removed throttleTimeMs for the case when THROTTLING_QUOTA_EXCEEDED exception is thrown in telemetry APIs. Done.